cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -571,6 +726,57 @@ struct Resources {
     stack_slots: Vec<(StackSlot, StackSize)>,
 }
 
+impl Resources {
+    fn is_last_block(&self, block: Block) -> bool {
+        self.blocks.last().map_or(false, |(b, _)| *b == block)
+    }
+
+    /// Returns [JumpTable]'s where all blocks are forward of `block`
+    fn forward_jump_tables(&self, builder: &FunctionBuilder, block: Block) -> Vec<JumpTable> {
+        // Unlike with the blocks below jump table targets are not ordered, thus we do need
+        // to allocate a Vec here.
+        self.jump_tables
+            .iter()
+            .filter(|jt| {
+                let blocks = builder.func.jump_tables[**jt].as_slice();
+                blocks.iter().all(|target| *target > block)
+            })
+            .copied()
+            .collect()
+    }
+
+    /// Partitions blocks at `block`
+    ///
+    /// The first slice includes all blocks up to and including `block`.
+    /// The second slice includes all remaining blocks.
+    fn partition_blocks(
+        &self,
+        block: Block,
+    ) -> (&[(Block, BlockSignature)], &[(Block, BlockSignature)]) {
+        let partition_point = self
+            .blocks
+            .iter()
+            .enumerate()
+            .find(|(_, (b, _))| *b > block)
+            .map_or(self.blocks.len(), |(i, _)| i);
```

Comment:
Yeah, that's a great suggestion! I was able to get this to the point where we directly index into the blocks vector using the block number. 

We don't necessarily need to store the block alongside the Signature, however removing it makes it so that we can't return a slice here, and the caller of this is quite messy without it and without allocating anything.

Reply:
Oh, good point! Storing the block number does make things easier in the caller.